jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -156,7 +156,7 @@ public int read(byte[] b, int off, int len) throws IOException
                 long period=System.nanoTime()-_firstByteTimeStamp;
                 if (period>=1000)
                 {
-                    double data_rate = _contentArrived / ((1.0*period)/TimeUnit.SECONDS.toNanos(1));
+                    double data_rate = _contentArrived / (((double)period)/TimeUnit.SECONDS.toNanos(1));
```

Comment:
Simone,  I've not seen your version?  where is it?

The 1000 magic number is there to not apply this data rate until some time has passed, otherwise it will not be accurate.   But yes - it should not be a magic number.

The reason I'm using double, it to protect against numerical sensitivity.  If you divide by whole seconds, then there can be a big difference in the computed data rate between 1 and 2 seconds, specially if the actual time was 1.001 or 2.999 seconds.